src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const stateResult = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        newState = { ...newState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        newState = { ...newState };
+
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
```

Comment:
[CODE STYLE]: switch/case should always have default case for error handling. It's good to add a default case that handles unexpected action types. This can help to debug the code in the future.